packages/truffle-db/src/loaders/artifacts/test/index.ts

```diff
@@ -149,46 +165,150 @@ query getWorkspaceCompilation($id: ID!) {
+  });
 
-    const {
-      data: {
-        workspace: {
-          compilation: {
-            compiler: {
-              version
-            },
-            sources,
-            contracts
+  it("loads contract sources", async () => {
+    for(let index in sourceIds) {
+      let {
+        data: {
+          workspace: {
+            source: {
+              contents,
+              sourcePath
+            }
           }
         }
-      }
-    } = await db.query(GetWorkspaceCompilation, { id: expectedId });
-
-    expect(version).toEqual(Build[0].compiler.version);
-    expect(sources.length).toEqual(3);
-    for(let source in sources) {
-      expect(sources[source]["id"]).toEqual(sourceIds[source]["id"]);
-      expect(sources[source].contents).toEqual(Build[source].source);
-      expect(contracts[source].name).toEqual(Build[source].contractName);
+      } = await db.query(GetWorkspaceSource, sourceIds[index]);
+
+      expect(contents).toEqual(Build[index].source);
+      expect(sourcePath).toEqual(Build[index].sourcePath);
+    }
+  });
+
+  it("loads bytecodes", async () => {
+    for(let index in bytecodeIds) {
+      let {
+        data: {
+          workspace: {
+            bytecode: {
+              bytes
+            }
+          }
+        }
+      } = await db.query(GetWorkspaceBytecode, bytecodeIds[index]);
+
+      expect(bytes).toEqual(Build[index].bytecode);
+
     }
+  });
+
+  it("loads contracts", async () => {
+    let contractIds = [];
+
+    for(let index in Build) {
+      let expectedId = generateId({
+        name: Build[index].contractName,
+        abi: { json: JSON.stringify(Build[index].abi) },
+        sourceContract: { index: Build[index].compiler.name === "solc" ? +index : 0},
+        compilation: {
+          id: Build[index].compiler.name === "solc" ? expectedSolcCompilationId : expectedVyperCompilationId
+        }
+      });
 
-  })
+      contractIds.push({ id: expectedId });
+      let {
+        data: {
+          workspace: {
+            contract: {
+              id,
+              name,
+              abi: {
+                json
+              },
+              constructor: {
+                createBytecode: {
+                  bytes
+                }
+              },
+              sourceContract: {
+                source: {
+                  contents,
+                  sourcePath
+                }
+              },
+              compilation: {
+                compiler: {
+                  version
+                },
+                sources,
+                contracts
+              }
+            }
+          }
+        }
+      } = await db.query(GetWorkspaceContract, contractIds[index]);
+
+      expect(name).toEqual(Build[index].contractName);
+      expect(bytes).toEqual(Build[index].bytecode);
+      expect(contents).toEqual(Build[index].source);
+      expect(version).toEqual(Build[index].compiler.version);
+      expect(id).toEqual(contractIds[index].id);
+    }
+  });
```

Comment:
All-in-all, diggin' these tests! Reading them provides good assurance that the behavior is thoroughly tested, which is great! Nice job!